www/js/draw-model.js

```diff
@@ -0,0 +1,155 @@
+define([
+	"dojo/_base/declare",
+	"dojo/_base/array",
+	"dojo/_base/lang",
+	"dojo/dom-attr",
+	"dojo/dom-construct",
+	"./graph-objects",
+	"jsPlumb/jsPlumb",
+], function(declare, array, lang, attr, domConstruct, graphObjects){
+	return declare(null, {
+		_instance: null,
+		_model: null,
+		_colors:[
+			"#00ffff", "#f0e68c", "#add8e6", "#e0ffff", "#90ee90", "#d3d3d3", "#ffb6c1",
+			"#ffffe0", "#00ff00", "#008000", "#f5f5dc", "#0000ff", "#a52a2a", "#00ffff",
+			"#ffd700", "#4b0082", "#ff00ff", "#800000", "#000080", "#808000", "#ffa500",
+			"#c0c0c0", "#ffc0cb", "#ff0000", "#800080", "#ffff00", "#00008b", "#008b8b",
+			"#f0ffff", "#a9a9a9", "#006400", "#bdb76b", "#8b008b", "#556b2f", "#ff8c00",
+			"#9932cc", "#8b0000", "#e9967a", "#9400d3", "#ff00ff"
+		],
+		_borderColor: 39,
+		_backgroundColor: 0,
+
+		constructor: function(model){
+			this._model = model;
+			this.connectorUI = {
+				endpoint: ["Dot", {radius: 2}],
+				endpointStyle: {fill: "#5c96bc"}
+			};
+			var instance = jsPlumb.getInstance({
+				Endpoint: ["Dot", {radius: 2}],
+				HoverPaintStyle : {strokeStyle:"#1e8151", lineWidth:2 },
+				Container:"statemachine-demo"
+			});
+
+			this._instance = instance;
+			this._model = model;
+			var vertices = [];
+			var temp = []
+			vertices = array.map(model.getNodes(), function(node){
+				temp = temp.concat(this.addNode(node));
+				return temp;
+			}, this);
+			vertices = vertices[vertices.length - 1]; //hack for keeping it one dimension
+			console.log(vertices);
```

Comment:
This tells the flow when we are in development. It will be useful. The build should take care of all this in production.